Source/Utility/UDim4.lua

```diff
@@ -1,34 +1,124 @@
-local UDim4 = {}
-UDim4.__index = UDim4
+-- @author MemoryPenguin
+-- @author Narrev
+-- UDim4 Object
 
-function UDim4.new(x, y)
-	assert(x and y, "Must supply x and y values.")
-	
-	local self = setmetatable({
-		X = x;
-		Y = y;
-	}, UDim4)
-	
-	return self
+local newVector2 = Vector2.new
+local newUDim2 = UDim2.new
+local IsA = game.IsA
+local SizePos = {
+	Size = "AbsoluteSize";
+	Position = "AbsolutePosition";
+}
+
+local function Connect(Event, func)
+	Connect = Event.Connect
+	Connect(Event, func)
+end
+
+local function Disconnect(Event)
```

Comment:
Same issue as Connect: I would prefer not to use helper functions just to microoptimize a member lookup.